cms/models.py

```diff
@@ -744,6 +771,98 @@ def get_context(self, request, *args, **kwargs):
         }
 
 
+class ExternalCoursePage(ProductPage):
+    """
+    CMS page representing an external course
+    """
+
+    template = "product_page.html"
+
+    parent_page_types = ["CourseIndexPage"]
+
+    external_url = models.URLField(
+        null=False, blank=False, help_text="The URL of the external course web page."
+    )
+    readable_id = models.CharField(
+        max_length=64,
+        null=False,
+        blank=False,
+        unique=True,
+        help_text="The readable ID of the external course. Appears in URL, has to be unique.",
+    )
+    start_date = models.DateField(
+        null=True, blank=True, help_text="The start date of the external course."
+    )
+    price = models.DecimalField(
+        null=True,
+        blank=True,
+        decimal_places=2,
+        max_digits=20,
+        help_text="The price of the external course.",
+    )
+
+    content_panels = [
+        FieldPanel("external_url"),
+        FieldPanel("readable_id"),
+        FieldPanel("start_date"),
+        FieldPanel("price"),
+    ] + ProductPage.content_panels
+
+    def get_url_parts(self, request=None):
+        # We need to skip `ProductPage` in the MRO and get to `Page.get_url_parts`
+        # pylint: disable=bad-super-call
+        url_parts = super(ProductPage, self).get_url_parts(request=request)
+        if not url_parts:
+            return None
+        return (
+            url_parts[0],
+            url_parts[1],
+            # Wagtail generates the 'page_path' part of the url tuple with the
+            # parent page slug followed by this page's slug (e.g.: "/courses/my-page-title").
+            # We want to generate that path with the parent page slug followed by the readable_id
+            # of the external course instead (e.g.: "/courses/external:some+external+course")
+            re.sub(
+                self.slugged_page_path_pattern,
+                r"\1{}\3".format(self.readable_id),
+                url_parts[2],
+            ),
```

Comment:
can we add tests for this method? particularly because we are using indexed and regex.